### ISSUE TITLE
AWS SDK v3 support

### ISSUE THREAD

**Issue by air2** [ArsenyYankovsky/typeorm-aurora-data-api-driver#148]
Hello is there any plan to migrate to AWS SDK v3?
I saw already a fork created by @Almaju, which uses SDK v3. 

If there is a plan to move to SDK v3, what would be the expected timeframe?

**Comment by KennethWussmann** [comment]
The driver is incompatible with Node 18 on AWS Lambda out-of-the-box now, because the Node 18 runtime only bundles with AWS SDK v3.

As the package also [doesn't bundle the aws-sdk](https://github.com/ArsenyYankovsky/typeorm-aurora-data-api-driver/blob/master/rollup.config.ts#L19), one would have to install the old aws-sdk v2 themselves and bundle it with the lambda code.

I'd love to see the package being updated to AWS SDK v3 as well.

**Comment by vicary** [comment]
Upstream dependency needs to upgrade AWS SDK, tracking issues below:

1. jeremydaly/data-api-client#127
2. jeremydaly/data-api-client#89

I think they need help.

**Comment by webjay** [comment]
AWS says:

```
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!                                                                                                                  !!
!!  Node 16 is approaching end-of-life and will no longer be supported in new releases after 2023-09-11.            !!
!!  Please upgrade to a supported node version as soon as possible.                                                 !!
!!                                                                                                                  !!
!!  This software is currently running on node v16.20.0.                                                            !!
!!  As of the current release of this software, supported node releases are:                                        !!
!!  - ^18.0.0 (Planned end-of-life: 2025-04-30)                                                                     !!
!!  - ^16.3.0 (Planned end-of-life: 2023-09-11) [DEPRECATED]                                                        !!
!!                                                                                                                  !!
!!  This warning can be silenced by setting the JSII_SILENCE_WARNING_DEPRECATED_NODE_VERSION environment variable.  !!
!!                                                                                                                  !!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

**Comment by webjay** [comment]
I have upgraded to Node 18 by having a layer with:
```
  "dependencies": {
    "aws-sdk": "2.1457.0",
    "pg": "8.11.3",
    "typeorm": "0.3.17",
    "typeorm-aurora-data-api-driver": "2.4.4"
  }

```

**Comment by ArsenyYankovsky** [comment]
I removed the `data-api-client` as a dependency and migrated to AWS SDK V3. Included in the new `3.0.0` version that was just released.
